PhysicsTools/PatAlgos/plugins/PATIsolatedTrackProducer.cc

```diff
@@ -1,92 +1,377 @@
+        bool isInPackedCands = (pcref.isNonnull() && pcref.id()==pc_h.id() && pcref.get()->charge()!=0);
+        bool isInLostTracks  = (ltref.isNonnull() && ltref.id()==lt_h.id());
+
+        LorentzVector p4;
+        pat::PackedCandidateRef refToCand;
+        int pdgId, charge, fromPV;
+        float dz, dxy, dzError, dxyError;
+        int pfCandInd; //to avoid counting packedPFCands in their own isolation
+
+        // get the four-momentum and charge
+        if(isInPackedCands){
+            p4 = pcref.get()->p4();
+            charge = pcref.get()->charge();
+            pfCandInd = pcref.key();
+        }else if(isInLostTracks){
+            p4 = ltref.get()->p4();
+            charge = ltref.get()->charge();
+            pfCandInd = -1;
+        }else{
+            double m = 0.13957018; //assume pion mass
+            double E = sqrt(m*m + gentk.p()*gentk.p());
+            p4.SetPxPyPzE(gentk.px(), gentk.py(), gentk.pz(), E);
+            charge = gentk.charge();
+            pfCandInd = -1;
+        }
+
+        if(p4.pt() < pT_cut_)
+            continue;
+        if(charge == 0)
+            continue;
+
+        // get the isolation of the track
+        pat::PFIsolation isolationDR03;
+        pat::PFIsolation miniIso;
+        getIsolation(p4, pc, pfCandInd, isolationDR03, miniIso);
+        
+        // isolation cut
+        if( p4.pt() < pT_cut_noIso_ && 
+            !(isolationDR03.chargedHadronIso() < absIso_cut_ ||
+              isolationDR03.chargedHadronIso()/p4.pt() < relIso_cut_ ||
+              miniIso.chargedHadronIso()/p4.pt() < miniRelIso_cut_))
+            continue;
+
+        // get the rest after the pt/iso cuts. Saves some runtime
+        if(isInPackedCands){
+            pdgId = pcref.get()->pdgId();
+            dz = pcref.get()->dz();
+            dxy = pcref.get()->dxy();
+            dzError = pcref.get()->dzError();
+            dxyError = pcref.get()->dxyError();
+            fromPV = pcref.get()->fromPV();
+            refToCand = pcref;
+        }else if(isInLostTracks){
+            pdgId = 0;
+            dz = ltref.get()->dz();
+            dxy = ltref.get()->dxy();
+            dzError = ltref.get()->dzError();
+            dxyError = ltref.get()->dxyError();
+            fromPV = -1;
+            refToCand = ltref;
+        }else{
+            pdgId = 0;
+            dz = gentk.dz();
+            dxy = gentk.dxy();
+            dzError = gentk.dzError();
+            dxyError = gentk.dxyError();
+            fromPV = -1;
+            refToCand = pat::PackedCandidateRef();  //NULL reference
+        }
+
+        float caloJetEm, caloJetHad;
+        getCaloJetEnergy(p4, caloJets.product(), caloJetEm, caloJetHad);
+
+        // if no dEdx info exists, just store -1
+        float dEdxPixel=-1, dEdxStrip=-1;        
+        if(gt2dedxHitInfo.isValid()){
+            const reco::DeDxHitInfo* hitInfo = (*gt2dedxHitInfo)[tkref].get();
+            dEdxPixel = getDeDx(hitInfo, true, false);
```

Comment:
pixel DeDxData is now available in the release. Please use it instead of this method.

For compatibility with 80X, please leave this setup (configured off by default) to read from DeDxHitInfo.

This way in 9X version we will get pixel dedx for more tracks (instead of just pt>10).